src/config/cli_config.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <boost/program_options.hpp>
+#include <string>
+
+#include "config/config.hpp"
+
+namespace taraxa {
+
+class CliConfig {
+ public:
+  CliConfig(int argc, const char* argv[], const std::string& version);
+
+  ~CliConfig() {}
+
+  // Returns true if node configuration is loaded successfully and command is node
+  bool nodeConfigured();
+
+  // Retrieves loaded node configuration
+  std::shared_ptr<FullNodeConfig> getNodeConfiguration();
+
+  enum NetworkIdType { Mainnet = 1, Testnet, Devnet };
```

Comment:
not sure if there are some specific limitations why enum is used here but in general enum class is preferred...